### PR TITLE
fix: Add condition check on Clear Filter

### LINKED ISSUE #2433: Clear Filter gets highlighted on No Param Passes
**Describe the bug**

When there are no parameters in search bar on startup page and After clicking search , Clear Filter gets highlighted .

**To Reproduce**

1. Go to 'Start-Up Page'
2. Click on 'Search'

**Expected behavior**

Unless some parameter is passed ,Clear-Filter should not get highlighted.

**Screenshots**

Screenshot 2019-03-31 at 5 15 44 AM

**Desktop (please complete the following information):**
 - OS: Mac OS Mojave
 - Browser: Chrome
 - Version: 73.0.3

### PR DESCRIPTION
**Pull request by naman360**
Fixes #2433

#### Short description of what this resolves:
It solves the issue of ClearFliter button being highlight when no param is passed.

#### Changes proposed in this pull request:

- Added a condition in `search()` function that checks whether a param or a filter is applied.

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: app/components/quick-filter.js [hunk 1]
```diff
@@ -74,7 +74,11 @@ export default class QuickFilter extends Component {
     this.setDateFilter();
     this.set('location', this.dummyLocation);
     this.set('eventName', this.dummyName);
-    this.set('disableClear', false);
+    if (this.dummyName || this.dummyLocation || this.startDate !== null) {
+      this.set('disableClear', false);
+    } else {
+      this.set('disableClear', true);
+    }
```
Comment (iamareebjamal) [inline app/components/quick-filter.js:81]: No need of if else, pass the boolean condition as second argument

**Comment by iamareebjamal** [comment]
Please follow project convention for PR title

File: app/components/quick-filter.js (hunk 1 above)
Comment (naman360) [reply app/components/quick-filter.js:81]: Can you please elaborate? I did not completely understood.

Comment (iamareebjamal) [reply app/components/quick-filter.js:81]: Move the condition from if statement to this.set call and remove the if statement

Comment (naman360) [reply app/components/quick-filter.js:81]: Got it.

File: app/components/quick-filter.js [hunk 2]
```diff
@@ -74,11 +74,7 @@ export default class QuickFilter extends Component {
     this.setDateFilter();
     this.set('location', this.dummyLocation);
     this.set('eventName', this.dummyName);
-    if (this.dummyName || this.dummyLocation || this.startDate !== null) {
-      this.set('disableClear', false);
-    } else {
-      this.set('disableClear', true);
-    }
+    this.set('disableClear', (this.dummyName || this.dummyLocation || this.startDate !== null) ? false : true);
```
Comment (iamareebjamal) [inline app/components/quick-filter.js:77]: No need for ternary operator

**Comment by iamareebjamal** [comment]
PR title is still not according to guidelines

**Comment by naman360** [comment]
Can I know from where can I read those guidlines.

**Comment by iamareebjamal** [comment]
See merged PRs titles and read about semantic commit messages

**Pull request merged**